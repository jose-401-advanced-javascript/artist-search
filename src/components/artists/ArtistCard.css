.ArtistCard {
  display: flex;
  flex-direction: column;
  background-color: lightslategray;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  padding: 1em 0em;
}