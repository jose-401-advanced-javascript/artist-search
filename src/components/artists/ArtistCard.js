import React from 'react';
import PropTypes from 'prop-types';
import styles from './ArtistCard.css';
import { Link } from 'react-router-dom';


const ArtistCard = ({ id, name, country, tags }) => {

  const tagElements = tags.map((tag, i) => {
    return (
      <p key={i}>{tag}</p>
    );
  });

  return (
    <div className={styles.ArtistCard}>

      <Link to={`/albums/${id}/${name}`}>
        <p>{name}</p>
      </Link>
      <p>{country}</p>
      {tagElements}
    </div>
  );
};

ArtistCard.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  country: PropTypes.string.isRequired,
  tags: PropTypes.arrayOf(PropTypes.object)
};

export default ArtistCard;
