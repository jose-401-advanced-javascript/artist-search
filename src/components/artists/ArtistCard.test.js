import React from 'react';
import { shallow } from 'enzyme';
import ArtistCard from './ArtistCard';

describe('ArtistCard component', () => {
  it('renders ArtistCard', () => {
    const wrapper = shallow(<ArtistCard id={'adsg'} country={'adsg'} name={'adsg'} tags={[
      {
        'count': 18,
        'name': 'rock'
      },
      {
        'count': 0,
        'name': 'progressive rock'
      },
      {
        'count': -1,
        'name': '70s'
      },
      {
        'count': -1,
        'name': '80s'
      },
    ]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
