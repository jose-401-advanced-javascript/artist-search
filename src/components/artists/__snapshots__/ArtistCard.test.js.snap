// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistCard component renders ArtistCard 1`] = `
<div
  className="ArtistCard"
>
  <Link
    to="/albums/adsg/adsg"
  >
    <p>
      adsg
    </p>
  </Link>
  <p>
    adsg
  </p>
  <p
    key="0"
  >
    <Component />
  </p>
  <p
    key="1"
  >
    <Component />
  </p>
  <p
    key="2"
  >
    <Component />
  </p>
  <p
    key="3"
  >
    <Component />
  </p>
</div>
`;
